### PR TITLE
CI: Pinning flake8-rst, last version raises incorrect errors

### LINKED ISSUE #24755: CI: azure failing
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=6880

```
doc/source/comparison_with_sas.rst(754,-22): error F821: undefined name 'get_ipython'
Bash exited with code '1'.
```

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] closes #24755
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

CC: @jreback @FHaase

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ dependencies:
   - cython>=0.28.2
   - flake8
   - flake8-comprehensions
-  - flake8-rst>=0.6.0
```
Comment (jreback) [inline environment.yml]: just make >=0.6.0<=0.7.0 for now

**Comment by jreback** [comment]
cool. merge on pass.

**Comment by jreback** [comment]
```
requirements-dev.txt(,): error : `requirements-dev.txt` has to be generated with `ci/../scripts/generate_pip_deps_from_conda.py` after `environment.yml` is modified.
Bash exited with code '1'.
```

**Comment by datapythonista** [comment]
Just saw it, luckily the syntax for >= and <= seems to be the same in pip as in conda, will push in a second, after I finish testing.

**Pull request merged**